pkg/pgmodel/cache/cache.go

```diff
@@ -51,26 +56,68 @@ type MetricNameCache struct {
 	Metrics *clockcache.Cache
 }
 
+func (m *MetricNameCache) key(schema, metric string) (int, string) {
+	// deep copy the strings so the original memory doesn't need to stick around
+	keyBuilder := strings.Builder{}
+	keyBuilder.Grow(len(metric) + len(schema) + len(fieldSeparator))
+	keyBuilder.WriteString(schema)
+	keyBuilder.WriteString(fieldSeparator)
+	keyBuilder.WriteString(metric)
+	return keyBuilder.Len(), keyBuilder.String()
```

Comment:
why do we need the len if we are returning the string anyway? cant we just use `len`